File: client/src/schemas/changes.ts
export interface ChangeResponseSchema {
    id: number;
    ip: string;
    community: string;
    sysname: string;
    ifIndex: number;
    oldInterface: ChangeInterfaceSchema;
    newInterface: ChangeInterfaceSchema;
    operator: string | null;
}

export interface ChangeInterfaceSchema {
    id: number;
    equipment: number;
    date: string;
    type: string;
    ifName: string;
    ifDescr: string;
    ifAlias: string;
    ifHighSpeed: number;
    ifOperStatus: string;
    ifAdminStatus: string;
}

File: server/src/schemas/responses/changes.py
from pydantic import BaseModel

class OldInterfaceSchema(BaseModel):
    """Schema of the old interface."""

    id: int
    date: str
    ifName: str
    ifDescr: str
    ifAlias: str
    ifHighSpeed: int
    ifOperStatus: str
    ifAdminStatus: str
                                       

class NewInterfaceSchema(BaseModel):
    """Schema of the new interface."""

    id: int
    date: str
    ifName: str
    ifDescr: str
    ifAlias: str
    ifHighSpeed: int
    ifOperStatus: str
    ifAdminStatus: str

class ChangesResponse(BaseModel):
    """Schema of the changes."""
    id: int
    ip: str
    community: str
    sysname: str
    ifIndex: int
    oldInterface: OldInterfaceSchema
    newInterface: NewInterfaceSchema
    assigned: bool
